src/client.py

```diff
@@ -16,33 +25,31 @@ def client(message):
     message += '\r\n'
 
     if version_info[0] == 2:
-        print('python 2')
         message = message.decode("utf8")
 
     print("Sending: ", message.encode('utf8'))
     client.sendall(message.encode('utf8'))
 
-    buffer_length = 8
-    result = u""
+    result = []
+
+    while b"\r\n" not in b''.join(result):
```

Comment:
Since `in` and `str.join` are both O(n) calls, and you're doing this every loop, this is very inefficient. I'm not reducing grade based on this, just critiquing :)

A better approach would be to just check each chunk as it comes in to see if it endswith `\r\n`.